docs/DeveloperGuide.adoc

```diff
@@ -46,177 +39,186 @@ The following class plays an important role at the architecture level:
-* does not depend on any of the other three components.
 
-[NOTE]
-As a more OOP model, we can store a `Tag` list in `Address Book`, which `Person` can reference. This would allow `Address Book` to only require one `Tag` object per unique `Tag`, instead of each `Person` needing their own `Tag` object. An example of how such a model may look like is given below. +
- +
-image:BetterModelClassDiagram.png[]
+.`FlashCard`,
+
+* There are 3 types of cards used by our App, the `JavascriptCard`, `FrontBackCard` and `MultipleChoiceCard`.
+* `JavascriptCard` and `FrontBackCard` implements from `FlashCard`, an interface class.
+* `MultipleChoiceCard` inherits from `FrontBackCard`.
+* The interface class `FlashCard` also implements `JsonInterface` class to allow loading and saving to Json format.
+
+.FlashCard Priority
+Each FlashCard has an associated priority level, which the user can indicate his/her value of the importance of the card.
+
+The priority levels are described below:
+
+[width="59%",cols="22%,<23%,<25%",options="header",]
+|=======================================================================
+| Priority| Value| Intended Action
+
+| HIGH_PRIORITY| 10| Valued as important by user, should be tested more often
+
+| LOW_PRIORITY| 1| Valued, but less important by user, should be tested less often
+
+|=======================================================================
+
+The sequence diagram below shows how the `ExamRunner` component can use the `createSubsetForTest()` method in the `Deck` and make use of the priorities associated with the `FlashCards` to create a test that emphasises the `FlashCards` that the user deems as important.
+
+image::flashcards_priority cards.png[]
+
+* Constructor of `ExamRunner` is called.
+* In the constructor of `ExamRunner`, a method call is made to `Deck` to create a subset of cards for test.
+* `Deck` processes the method call, responds as such:
+** If there are less than 10 cards in `Deck`, return all the cards.
+** If there are more than 10 cards in `Deck`, return a set of cards with both `HIGH_PRIORITY` and `LOW_PRIORITY` cards, which the proportion is weighted at 60% to 40% ration respectively.
+* `Deck` returns the test set to `ExamRunner`.
+
+[Note]
+The current implementation has yet to handle the situation where `ExamRunner` creates a test set when there are no cards in Deck. This will be implemented before the next code release.
 
 [[Design-Storage]]
 === Storage component
 
 .Structure of the Storage Component
 image::StorageClassDiagram.png[]
 
-*API* : link:{repoURL}/src/main/java/seedu/address/storage/Storage.java[`Storage.java`]
+*API* : link:{repoURL}/src/main/java/dream/fcard/logic/storage/StorageManager.java[`StorageManager.java`]
+*API* : link:{repoURL}/src/main/java/dream/fcard/util/json/JsonParser.java[`JsonParser.java`]
+*API* : link:{repoURL}/src/main/java/dream/fcard/util/FileReadWrite.java[`FileReadWrite.java`]
 
 The `Storage` component,
```

Comment:
Exception should have a dependency arrow instead of an association arrow and the relation should be throws.